### ISSUE TITLE
Conflict of step def in 2.15 and 2.17.0.3-SNAPSHOT but working with 2.14

### ISSUE THREAD

**Issue by anshu781126** [gwen-interpreter/gwen-web#55]
Example:
Scenario: I "<columnmenuoption>" on "<gridname>" grid-column "<column_name>"
Scenario: I "<grid_title_operation>" on "<gridname>"

On executing this on 2.14 it was working as expected -
And I "Grouping" on "Exceptions" grid-column "Priority"

But on 2.15 and 2.17.0.3-SNAPSHOT it is giving conflict - 

ERROR - Failed step [at line 1]: And I "Grouping" on "Exceptions" grid-column "Priority": Ambiguous condition in resolving 'I "Grouping" on "Exceptions" grid-column "Priority"
': 1 StepDef match expected but 2 found: I "<columnmenuoption>" on "<gridname>" grid-column "<column_name>",I "<grid_title_operation>" on "<gridname>"
ERROR - [26ms] Failed Step: And I "Grouping" on "Exceptions" grid-column "Priority"

**Comment by bjuric** [comment]
Looks like some escaping issues in your comment. To confirm, you have..

Two StepDefs named with parameters structured like this..
```
@StepDef
Scenario: I "<a>" on "<b>" and "<c>"
...

@StepDef
Scenario: I "<a>" on "<b>"
...
```

Then in your feature, you call..

`Given "1" on "2" and "3"`

Gwen reports an ambiguous condition since it does actually match both of them as follows:

```
@StepDef
Scenario: I "<a>" on "<b>" and "<c>"

   <a>=1
   <b>=2
   <c>=3
```

And also..

```
@StepDef
Scenario: I "<a>" on "<b>"

   <a>=1
   <b>=2" and "3
```

Therefore this is an ambiguous condition that is reported correctly in recent Gwen releases. Previous releases of failed to detect this (and it was working for you by coincidence). So you need to rename one of your StepDefs to remove the ambiguity.

**Comment by anshu781126** [comment]
@bjuric  yes I have updated the comment with backticks. You rightly understood the scenario but I still feel gwen shouldn't go and search for stepdef before reading complete statement from the feature.

Let's considering your Scenarios --

@StepDef
Scenario: I "<a>" on "<b>" and "<c>"
...

@StepDef
Scenario: I "<a>" on "<b>"
...

both the scenarios defined by you are different. Can't gwen evaluate the statement only after reading it completely. As one of the scenario has `and "<c>"` followed by the common part between the two. That also makes the statement different.

**Comment by anshu781126** [comment]
When gwen starts processing say it has -

`I "A" on "B"`

it doesn't have the third argument then why it is difficult to identify for gwen; can't it refer stepdef 2 as it also  has only 2 arguments?

greedy regex is the limitation in this case?

**Comment by anshu781126** [comment]
We are using quotes for parameters to make it intuitive in feature file. But without quotes I believe it would be difficult for gwen to figure out how many arguments are present. Am I on same page?

**Comment by bjuric** [comment]
Quotes won't matter, the same ambiguity would be detected.

When Gwen starts processing, it could match either the 2-arg or 3-arg StepDef first. The order in which StepDefs are matched is non-determinant. 

When Gwen finds more that one match, how would it choose one over the other?
